### ISSUE TITLE
Panache LockModeType ignored since 3.1.1

### ISSUE THREAD

**Issue by jendib** [quarkusio/quarkus#33889]
### Describe the bug

When using `findById(id, LockModeType.PESSIMISTIC_WRITE)` the LockModeType seems to be ignored in the final query.

```java
Lucky lucky = Lucky.findById(luckyId, LockModeType.PESSIMISTIC_WRITE);
```

Quarkus 3.1.0;
```sql
    select
        l1_0.id,
        l1_0.createdDate,
        l1_0.expiringDate
    from
        Lucky l1_0 
    where
        l1_0.id=? for no key update
```

Quarkus 3.1.1:
```sql
    select
        l1_0.id,
        l1_0.createdDate,
        l1_0.expiringDate
    from
        Lucky l1_0 
    where
        l1_0.id = any (?)
```

### Expected behavior

`for no key update` should be added.

### Output of `java -version`

17

### Quarkus version or git rev

3.1.1

### Additional information

PostgreSQL dialect.

**Comment by FroMage** [comment]
This looks like an ORM bug, since we just pass it on, no? @yrodiere?

**Comment by yrodiere** [comment]
Most likely, yes. This might also just be a misconfigured [dialect or database version](https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.0:-Hibernate-ORM-5-to-6-migration#dialect), since not all dialects support all locks, and dialect configuration changed in Quarkus 3 / ORM 6.

@jendib can you please provide a reproducer?

**Comment by jendib** [comment]
@yrodiere I attached a reproducer.
- Run the test and see the SQL query in logs (without `for update`)
- Change the Quarkus version to 3.1.0 in pom.xml and re-run the test => `for update` is present

[reproducer.zip](https://github.com/quarkusio/quarkus/files/11691130/reproducer.zip)

**Comment by yrodiere** [comment]
Thanks!

Hey @marko-bekhta could you please have a look? From what I can see `org.hibernate.orm.test.jpa.lock.LockTest#testFindWithPessimisticWriteLockTimeoutException` passes just fine in the Hibernate ORM test suite on 6.2.4, so I wonder what's different here.

**Comment by marko-bekhta** [comment]
@yrodiere, soooo in short -- the difference is in batch fetching. 
Disabling it in Quarkus (3.1.1) with `quarkus.hibernate-orm.fetch.batch-size=-1` will result in generating an expected query:
```sql
    select
        m1_0.id,
        m1_0.field 
    from
        MyEntity m1_0 
    where
        m1_0.id=? for no key update
```

The problem was introduced in  ORM 6.2.2  but is already fixed in the main branch (6.3)

**Comment by yrodiere** [comment]
Alright, thanks for looking into this.

Keeping this open until we upgrade to a version of ORM that fixes the bug.

Can you please give us a link to the ORM issue, @marko-bekhta ?

**Comment by joachimglink** [comment]
I came across the same issue today when upgrading from Quarkus 3.1.0 to 3.1.1

But I don´t really see why I should disable the fetch batch-size. After all, I do a Panache #findById call where I always expect to get 0..1 entities. So why having a batch loading functionality in this case at all?
Or do I miss something here?

**Comment by jendib** [comment]
@marko-bekhta Could you please give us the associated Hibernate ticket? Thanks.

**Comment by marko-bekhta** [comment]
Sure, this particular case with lock mode is going to be fixed under https://hibernate.atlassian.net/browse/HHH-16820.

**Comment by jendib** [comment]
It should be fixed with https://github.com/quarkusio/quarkus/pull/34441

**Comment by yrodiere** [comment]
> It should be fixed with #34441

Right, thanks. Closing.
